src/pages/auth/Signup.jsx

```diff
@@ -0,0 +1,250 @@
+import React, { useState } from 'react';
+import { Button, Divider, Flex, Form, Input, Spin } from 'antd';
+import { FaRegEdit } from 'react-icons/fa';
+import { Link, useNavigate } from 'react-router-dom';
+import { authService } from '../../services';
+import { validateObject, handleConfirmPassword } from '../../utils';
+import { PATH_LOGIN } from '../../constants/routes';
+import {
+    ADDRESS_MESSAGE,
+    CONTACT_NUMBER_MESSAGE,
+    FIRST_NAME_MESSAGE,
+    LAST_NAME_MESSAGE,
+    PASSWORD_MESSAGE,
+    SUCCESSFUL_SIGNUP_MESSAGE,
+    USERNAME_MESSAGE,
+} from '../../constants/auth';
+import { toast } from 'react-toastify';
+
+const Signup = () => {
+    const navigate = useNavigate();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [form] = Form.useForm();
+
+    const handleSubmitForm = async (values) => {
+        setIsSubmitting(true);
+
+        const validation_errors = validateObject(values);
```

Comment:
Does the global validations applicable to this one or we need separate validations?